server/pulp/server/db/model/__init__.py

```diff
@@ -399,45 +409,44 @@ def pre_save_signal(cls, sender, document, **kwargs):
         document.last_updated = dateutils.now_utc_timestamp()
 
         # If content was set on this unit, copy the content into place
-        if document._source_location and document._relative_path:
+        if document._source_location:
             server_storage_dir = config.config.get('server', 'storage_dir')
-            platform_storage_dir = os.path.join(server_storage_dir, 'content',
-                                                document.unit_type_id)
-            target_location = os.path.join(platform_storage_dir, document._relative_path)
+            platform_storage_location = os.path.join(server_storage_dir, 'units',
+                                                     document.unit_type_id,
+                                                     str(document.id)[0],
+                                                     str(document.id)[1:3],
+                                                     str(document.id))
             # Make if source is a directory, recursively copy it, otherwise copy the file
             if os.path.isdir(document._source_location):
-                shutil.copytree(document._source_location, target_location)
+                shutil.copytree(document._source_location, platform_storage_location)
             else:
-                check_dir = os.path.dirname(target_location)
+                target_file_name = os.path.basename(document._source_location)
+                # Make sure the base directory exists
                 try:
-                    os.makedirs(check_dir)
+                    os.makedirs(platform_storage_location)
                 except OSError as e:
                     if e.errno != errno.EEXIST:
                         raise
-                shutil.copy(document._source_location, target_location)
+                # Copy the file
+                platform_storage_location = os.path.join(platform_storage_location,
```

Comment:
This should be named a different variable name. Maybe `unit_full_storage_location` or something similar. Generally, a variable should not be modified and saved back to to the same variable with the same variable.